sdk/state/snapshot_test.go

```diff
@@ -0,0 +1,248 @@
+	{
+		dtx, _, err := localChannel.CloseTxs()
+		require.NoError(t, err)
+		dtxXDR, err := dtx.Base64()
+		require.NoError(t, err)
+
+		successResultXDR, err := txbuildtest.BuildResultXDR(true)
+		require.NoError(t, err)
+
+		resultMetaXDR, err := txbuildtest.BuildResultMetaXDR([]xdr.LedgerEntryData{
+			{
+				Type: xdr.LedgerEntryTypeAccount,
+				Account: &xdr.AccountEntry{
+					AccountId: xdr.MustAddress(localEscrowAccount.Address()),
+					SeqNum:    102,
+					Signers: []xdr.Signer{
+						{Key: xdr.MustSigner(localSigner.Address()), Weight: 1},
+						{Key: xdr.MustSigner(remoteSigner.Address()), Weight: 1},
+					},
+					Thresholds: xdr.Thresholds{0, 2, 2, 2},
+				},
+			},
+			{
+				Type: xdr.LedgerEntryTypeAccount,
+				Account: &xdr.AccountEntry{
+					AccountId: xdr.MustAddress(remoteEscrowAccount.Address()),
+					SeqNum:    103,
+					Signers: []xdr.Signer{
+						{Key: xdr.MustSigner(remoteSigner.Address()), Weight: 1},
+						{Key: xdr.MustSigner(localSigner.Address()), Weight: 1},
+					},
+					Thresholds: xdr.Thresholds{0, 2, 2, 2},
+				},
+			},
+		})
+		require.NoError(t, err)
+
+		err = localChannel.IngestTx(dtxXDR, successResultXDR, resultMetaXDR)
+		require.NoError(t, err)
+		err = remoteChannel.IngestTx(dtxXDR, successResultXDR, resultMetaXDR)
+		require.NoError(t, err)
+
+		cs, err := localChannel.State()
+		require.NoError(t, err)
+		assert.Equal(t, StateClosing, cs)
+
+		cs, err = remoteChannel.State()
+		require.NoError(t, err)
+		assert.Equal(t, StateClosing, cs)
+	}
+
+	// Check snapshot rehydrates the channel identically when closing ingested.
+	assertChannelSnapshotsAndRestores(t, localConfig, localChannel)
+	assertChannelSnapshotsAndRestores(t, remoteConfig, remoteChannel)
+
+	// Confirm the close.
+	ca, err = remoteChannel.ConfirmClose(ca)
+	require.NoError(t, err)
+	_, err = localChannel.ConfirmClose(ca)
+	require.NoError(t, err)
+
+	// Check snapshot rehydrates the channel identically when payment confirmed.
+	assertChannelSnapshotsAndRestores(t, localConfig, localChannel)
+	assertChannelSnapshotsAndRestores(t, remoteConfig, remoteChannel)
+
+	// Put the channel into a close state by ingesting the close tx.
+	{
+		_, ctx, err := localChannel.CloseTxs()
+		require.NoError(t, err)
+		ctxXDR, err := ctx.Base64()
+		require.NoError(t, err)
+
+		successResultXDR, err := txbuildtest.BuildResultXDR(true)
+		require.NoError(t, err)
+
+		resultMetaXDR, err := txbuildtest.BuildResultMetaXDR([]xdr.LedgerEntryData{})
+		require.NoError(t, err)
+
+		err = localChannel.IngestTx(ctxXDR, successResultXDR, resultMetaXDR)
+		require.NoError(t, err)
+		err = remoteChannel.IngestTx(ctxXDR, successResultXDR, resultMetaXDR)
+		require.NoError(t, err)
+
+		cs, err := localChannel.State()
+		require.NoError(t, err)
+		assert.Equal(t, StateClosed, cs)
+
+		cs, err = remoteChannel.State()
+		require.NoError(t, err)
+		assert.Equal(t, StateClosed, cs)
+	}
```

Comment:
💭 can we add a final check? Since ingesting the closeTx could change agent/channel state
```suggestion
	}
	assertChannelSnapshotsAndRestores(t, localConfig, localChannel)
	assertChannelSnapshotsAndRestores(t, remoteConfig, remoteChannel)
```

Reply:
💯 Thanks for catching that.